crates/ty_python_semantic/src/site_packages.rs

```diff
@@ -390,6 +382,130 @@ System site-packages will not be used for module resolution.",
     }
 }
 
+/// A parser for `pyvenv.cfg` files: metadata files for virtual environments.
+///
+/// Note that a `pyvenv.cfg` file *looks* like a `.ini` file, but actually isn't valid `.ini` syntax!
+///
+/// See also: <https://snarky.ca/how-virtual-environments-work/>
+#[derive(Debug)]
+struct PyvenvCfgParser<'s> {
+    cursor: Cursor<'s>,
+    line_number: NonZeroUsize,
+    data: RawPyvenvCfg<'s>,
+}
+
+impl<'s> PyvenvCfgParser<'s> {
+    fn new(source: &'s str) -> Self {
+        Self {
+            cursor: Cursor::new(source),
+            line_number: NonZeroUsize::new(1).unwrap(),
+            data: RawPyvenvCfg::default(),
+        }
+    }
+
+    /// Parse the `pyvenv.cfg` file and return the parsed data.
+    fn parse(mut self) -> Result<RawPyvenvCfg<'s>, PyvenvCfgParseErrorKind> {
+        while !self.cursor.is_eof() {
+            self.parse_line()?;
+            self.line_number = self.line_number.checked_add(1).unwrap();
+        }
+        Ok(self.data)
+    }
+
+    /// Parse a single line of the `pyvenv.cfg` file and advance the cursor
+    /// to the beginning of the next line.
+    fn parse_line(&mut self) -> Result<(), PyvenvCfgParseErrorKind> {
+        let PyvenvCfgParser {
+            cursor,
+            line_number,
+            data,
+        } = self;
+
+        let line_number = *line_number;
+
+        cursor.eat_while(|c| c.is_whitespace() && c != '\n');
```

Comment:
I guess we don't support `\r` line endings. Seems reasonable. Does your code work correclty with `\r\n`?

Reply:
Added a test in https://github.com/astral-sh/ruff/pull/18057/commits/45e067a80da64c57da98f0060ba870c48ce9b18f that verifies this